platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/internal/EventImpl.java

```diff
@@ -106,6 +104,22 @@ public class EventImpl extends AbstractSerializableHashable
     /** The number of application transactions in this round */
     private int numAppTransactions = 0;
 
+    /**
+     * The sequence number of an event before it is added to the write queue.
+     */
+    public static final long NO_STREAM_SEQUENCE_NUMBER = -1;
+
+    /**
+     * The sequence number of an event that will never be written to disk because it is stale.
+     */
+    public static final long STALE_EVENT_STREAM_SEQUENCE_NUMBER = -2;
```

Comment:
we usually don't know that an event is stale until a lot of time has passed. by then, the event will probably already be written to disk

Reply:
I agree that this condition will not be triggered often during standard operations.

My code does not guarantee that stale events don't get written to disk. But if an event becomes stale before going to disk, my code will refuse to write it.

The intention of this constant is not to mark all stale events. This constant is used as a safeguard in case somebody attempts to wait for a stale event to be flushed to disk. Better to throw than to let somebody wait forever.

With this in mind, do you think this is an ok strategy? Or would you still like me to change it?

Reply:
its ok